File: algorithms_and_data_structures/lecture-1/find_simple_numbers.py
import time

# Поиск простых чисел в диапазоне

start = time.time()

# код программы


def find_simple_numbers():
    max = int(input("Введите максимальное число: "))
    result = []
    for i in range(1, max+1):
        simple = True
        for j in range(2, i):
            if (i % j == 0):
                simple = False
                break
        if simple:
            result.append(i)

    return result


print(find_simple_numbers())

end = time.time() - start

print(end)

File: algorithms_and_data_structures/lecture-1/find_simple_numbers.java
import java.util.ArrayList;
import java.util.List;

public class find_simple_numbers {
    public static void main(String[] args){
        System.out.println(findSimpleNumbers(20));
    }
    public static List<Integer> findSimpleNumbers(int max) {
        List<Integer> result = new ArrayList<>();
        for (int i = 1; i <= max; i++) {
            boolean simple = true;
            for (int j = 2; j < i; j++) {
                if (i % j == 0) {
                    simple = false;
                    break;
                }
            }
            if (simple) {
                result.add(i);
            }
        }
        return result;
    }
}
